parkhangs-webapp/src/features/modal/ModalMapDetail.js

```diff
@@ -1,69 +1,87 @@
 import React from 'react'
-import AddEventForm from 'components/AddEventForm'
+// import AddEventForm from '../components/AddEventForm'
 import { deleteEvent,fetchEventsById } from 'features/parks/parksSlice.js'
 import { connect } from 'react-redux'
 import 'features/modal/ModalMapDetail.css'
 import {unwrapResult} from '@reduxjs/toolkit'
 
 
 class ModalMapDetail extends React.Component {
-  
-    selectedPark = () => {
-        return this.props.parks.find(park => park.parkName === this.props.park)
-    }
-   
+
+    // this.props.park is the current park. Passed in as a prop. Not from store
+
     componentDidMount = async () => {
         let result =  await this.getEvents();
-        //console.log(result); 
-        //console.log(this.props.eventsById);
         console.log(this.props.error);
         if(this.props.error){
             throw this.props.error
         }
         
-    }
+    };
 
     getEvents = async () => {
         try{
-             const resultAction = await this.props.fetchEvents(this.selectedPark().id) //waits for promise
+             const resultAction = await this.props.fetchEvents(this.props.park._id);//waits for promise
```

Comment:
You're passing in the mongoobject id, hence the promise for events being rejected.

Reply:
This works when it's written as:
`const resultAction = await this.props.fetchEvents(this.props.park.parkId)`